controllers/cloud.redhat.com/namespaces.go

```diff
@@ -79,6 +87,8 @@ func CreateNamespace(ctx context.Context, cl client.Client, pool *crd.NamespaceP
 }
 
 func SetupNamespace(ctx context.Context, cl client.Client, pool crd.NamespacePool, ns string) error {
+	initialLabels["pool"] = pool.Name
```

Comment:
This is going to give us the problem we discussed a while back. We should ensure that we don't mutate that initial map, and instead iterate to copy.